### ISSUE TITLE
MD044: False positives on 'html' in URLs and common words like 'rest', 'less'

### ISSUE THREAD

**Issue by joshrotenberg** [joshrotenberg/mdbook-lint#339]
## Description

MD044 (proper-names) is flagging words that happen to match configured proper names but are used in different contexts:

1. **'html' in file extensions/URLs** - flagging `.html` in link URLs
2. **Common words** - flagging 'rest' (as in "the rest of"), 'less' (as in "less often")

## Examples

False positives from The Rust Book:

```
warning[MD044]: Proper name 'html' should be capitalized as 'HTML'
  --> appendix-01-keywords.md:62:36
     |
  62 | [union]: ../reference/items/unions.html
     |                                    ^^^^ proper-names
```

```
warning[MD044]: Proper name 'rest' should be capitalized as 'REST'
  --> appendix-02-operators.md:46:100
     |
  46 | | `variant(x, ..)` | "And the rest" pattern binding |
     |                                            ^^^ proper-names
```

```
warning[MD044]: Proper name 'less' should be capitalized as 'Less'
  --> appendix-05-editions.md:8:9
     |
   8 | changes less often; Rust releases smaller updates more frequently
     |         ^^^^ proper-names
```

## Suggested Fixes

1. **For 'html'**: Skip matches inside URLs or file paths (anything matching `\.[a-z]+$` pattern or within `[](url)` syntax)
2. **For common words**: The default proper names list should probably not include words that are commonly used in lowercase ('rest', 'less'). These are too noisy. Or require word boundaries that exclude common phrases like "the rest of", "less than"

## Found While Testing

Ran mdbook-lint against rust-lang/book - found 280 false positives for this rule.
